lib/reek/reek_exception.rb

```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+module Reek
+  #
+  # @public
+  #
+  # ReekException is the only exception that Reek should give back
+  # to the user in case something goes wrong.
+  class ReekException < RuntimeError
```

Comment:
I have seen this pattern in a couple of libraries (can't remember which though right now unfortunately) and I kind of like it since I find it very intention revealing. `Error` in the context of `Reek` doesn't tell me much as an outsider since it's pretty vague. `ReekException` (or maybe better: `ReekError`?) kind of signals that this is the exception `Reek` will propagate to the caller.